lib/tiddle/strategy.rb

```diff
@@ -57,15 +57,28 @@ def authentication_keys
       end
 
       def touch_token(token)
-        token.update_attribute(:last_used_at, Time.current) if token.last_used_at < 1.hour.ago
+        return unless token.last_used_at < touch_token_interval(token).ago
+
+        token.update_attribute(:last_used_at, Time.current)
       end
 
       def unexpired?(token)
-        return true unless token.respond_to?(:expires_in)
-        return true if token.expires_in.blank? || token.expires_in.zero?
+        return true if expiration_disabled?(token)
 
         Time.current <= token.last_used_at + token.expires_in
       end
+
+      def touch_token_interval(token)
+        return 1.hour if expiration_disabled?(token) || token.expires_in >= 2.hours
+
+        1.minute
```

Comment:
If you imagine a token with an expiration time of 15 minutes, one minute is more precise than 5 minutes. Because if the token is used in the first 4 minutes and back 11 minutes after, the token will be expired.

Also, the idea of 1 minute is that it supports all the short-term tokens, even a token with an expiration of 5 minutes that is hard to exist.

It is an hour or minute scale; depending on the token expiration time, we choose the one that fits better.